astropy/io/fits/header.py

```diff
@@ -1540,9 +1528,24 @@ def remove(self, keyword):
         keyword : str
             The keyword of which to remove the first instance in the header
 
+        ignore_missing : bool, optional
+            When True, ignores missing keywords.  Otherwise, if the keyword is not present
+            in the header a KeyError is raised.
+
+        all : bool, optional
+            When True, all instances of keyword will be removed.
+            Otherwise only the first instance of the given keyword is removed.
+
         """
+        keyword = Card.normalize_keyword(keyword)
+        if keyword in self._keyword_indices:
+            del self[self._keyword_indices[keyword][0]]
+            if all:
```

Comment:
`all` is maybe not a so good idea as argument name, as it shadows the builtin `all` function. Maybe `remove_all` ?